packages/experimental/assistant/src/ai-service/types.ts

```diff
@@ -34,25 +34,25 @@ export const GenerateRequest = S.Struct({
   model: S.optional(LLMModel),
 
   /**
-   * System instructions to the LLM.
+   * Tools available for the LLM.
    */
-  systemPrompt: S.optional(S.String),
+  tools: S.optional(S.Array(Tool).pipe(S.mutable)),
 
   /**
-   * Current request.
+   * System instructions to the LLM.
    */
-  // TODO(burdon): Remove from history.
-  // prompt: Message,
+  systemPrompt: S.optional(S.String),
 
   /**
    * History of messages to include in the context window.
    */
+  // TODO(burdon): Rename messages.
   history: S.optional(S.Array(Message)),
 
   /**
-   * Tools available for the LLM.
+   * Current request.
    */
-  tools: S.optional(S.Array(Tool).pipe(S.mutable)),
+  prompt: S.optional(Message),
```

Comment:
does this require changes on the service side to respect `promp`?

Reply:
not currently; to discuss how to evolve in common.